core/jmx/impl/src/main/java/org/opennms/netmgt/jmx/impl/connection/connectors/Jsr160ConnectionFactory.java

```diff
@@ -57,6 +65,11 @@
     
     private static final Logger LOG = LoggerFactory.getLogger(Jsr160ConnectionFactory.class);
 
+    // Set default timeout to 2 secs as specified in AbstractDetector
+    private static final long DEFAULT_TIMEOUT = 2000;
```

Comment:
This connection factory is also used by other components.

Can we have it use 2 seconds as the default timeout when invoked by a detector, and 30 seconds otherwise?